firedrake/interpolation.py

```diff
@@ -142,20 +143,12 @@ def _ufl_expr_reconstruct_(self, expr, v=None, **interp_data):
 
 
 @PETSc.Log.EventDecorator()
-def interpolate(
-    expr,
-    V,
-    subset=None,
-    access=op2.WRITE,
-    allow_missing_dofs=False,
-    default_missing_val=None,
-    ad_block_tag=None
-):
-    """Interpolate an expression onto a new function in V.
+def interpolate(expr, V, *args, **kwargs):
+    """Returns a UFL expression for the interpolation operation of ``expr`` into ``V``.
 
     :arg expr: a UFL expression.
     :arg V: the :class:`.FunctionSpace` to interpolate into (or else
-        an existing :class:`.Function`).
+        an existing :class:`.Function` or :class:`.Cofunction`).
```

Comment:
I thought `expr` could only be primal. If we want to interpolate a `Cofunction` (or a `BaseForm`) we can pass it to `V` as the second arg, and we would have to pass a `TestFunction` to `expr` in the first arg.

Reply:
Nevermind the docstring seems okay